app/main.py

```diff
@@ -99,28 +115,124 @@ def _decrypt(self, encrypted_jwt, tx_id):
 
 
+class SetHealth:
+    """This handles all the healthcheck functionality for the application.
+
+       The status of the application is determined by the rabbitmq health and ftp health.
+       This is done by performing a healthcheck on rabbitmq and checking the application
+       has a live ftp connection. This check is done in the background after a delay.
+       When the healthcheck endpoint endpoint is hit a 200 status is returned with
+       app status as well as the dependencies of rabbitmq and ftp."""
+
+    def __init__(self):
+        self.stop_flag = Event()
+        self.http_client = HTTPClient()
+        self.ftp = SDXFTP(logger,
+                          settings.FTP_HOST,
+                          settings.FTP_USER,
+                          settings.FTP_PASS,
+                          settings.FTP_PORT,
+                          )
+        self.conn = self.ftp._connect()
+        self.set_ftp_status()
+        self.set_rabbit_status()
+
+    def set_rabbit_status(self):
+        try:
+            resp = self.http_client.fetch(settings.RABBIT_HEALTHCHECK_URL,
+                                          auth_username=settings.SEFT_RABBITMQ_MONITORING_USER,
+                                          auth_password=settings.SEFT_RABBITMQ_MONITORING_PASS)
+            body = json.loads(resp.body.decode('utf8'))
+            status = body.get('status')
+            self.rabbit_status = False if not status else True
+        except HTTPError as e:
+            logger.error("Error receiving rabbit health " + str(e))
+
+    def set_ftp_status(self):
+        try:
+            self.conn.voidcmd("NOOP")
+            self.ftp_status = True
+        except Error as e:
+            logger.error("FTP error raised" + str(e))
+
+    def start(self):
+        thread = TimerThread(self.stop_flag, self.get_health())
+        thread.run()
+
+    def stop(self):
+        self.stop_flag.set()
+
+    def get_health(self):
+        self.set_rabbit_status()
+        self.set_ftp_status()
+
+        if self.rabbit_status and self.ftp_status:
+            self.app_health = True
+        else:
+            self.app_health = False
+
+
+class HealthCheck(tornado.web.RequestHandler):
+
+    def initialize(self):
+        self.set_health = SetHealth()
+
+    def get(self):
+        self.write({"status": self.set_health.app_health,
+                    "dependencies": {"rabbitmq": self.set_health.rabbit_status,
+                                     "ftp": self.set_health.ftp_status}})
+
+
+class TimerThread(Thread):
+
+    def __init__(self, event, p):
+        Thread.__init__(self)
+        self.stopped = event
+        self.p = p
+
+    def run(self):
+        while not self.stopped.wait(HEALTHCHECK_DELAY_SECONDS):
+            self.p.__call__()
+
+
+def make_app():
+    return tornado.web.Application([
+        (r"/healthcheck", HealthCheck),
+    ])
+
+
 def main():
     logger.debug("Starting SEFT consumer service")
 
+    set_health = SetHealth()
```

Comment:
Is SetHealth the right name for this?  I don't know what is the right name, but this doesn't seem right